profiler/profile.go

```diff
@@ -15,6 +15,7 @@ import (
 	"time"
 
 	"github.com/DataDog/gostackparse"
+	"github.com/felixge/pprofutils"
```

Comment:
Yeah, my initial idea was to copy the required code from pprofutils. The reason I didn't do it, is because the [test cases for delta profiles](https://github.com/felixge/pprofutils/blob/main/delta_test.go) rely on all the remaining functionality of the library (converting pprof files between text/protobufs), so the only viable approach would be to copy the entire library. Would you prefer that?

Reply:
I think it's fine and natural to copy the entire library, if the library is mostly meant to be used with this package. One question I have is: is this a library you'd expect others to use? If that's the case, should it be a sub-module with its own versioning? Or, should it simply be in the Data~D~dog org as its own library?

Otherwise, if others should not use it, then simply copying it into an `internal` place is fine. In this case, I think we can dumb it down a bit and perhaps don't need the CLIs, unless those are useful to you for other things like debugging.

Just my 2c.

Reply:
> Yeah, my initial idea was to copy the *required code* from pprofutils.

Oh, sorry. My bad. You said "_copy the required code_". That's probably the right thing to do. In an `internal` package. For some reason I thought you want to copy the whole repo 🤦🏻

Reply:
@gbbr seems like there might still be confusion here : p. Let me rephrase my previous comment:

- dd-trace-go just needs a small amount of code from pprofutils which can just be copied
- However, we probably also want to copy the test cases for this code ?
- The test cases however end up depending on pretty much everything else in pprofutils

This means we can either copy the entire thing, or copy the code we need without any test cases. I'll probably just copy the entire thing to an `internal` directory for now. I don't care if it gets out of data with upstream pprofutils, nor do I think we should worry about exposing any of this to our users as part of dd-trace-go (it's out of scope).

Reply:
I think copying the required code sounds good. It's odd that you need the entire thing to test it, but I'm not familiar with the specifics so fine by me. Maybe we can try and delete everything that isn't necessary for this change, and also, move everything (the "entire thing") that isn't the actual required code into `*_test.go` files so it's clear that it's there only for testing.